src/os/bluestore/AvlAllocator.h

```diff
@@ -30,8 +30,24 @@ struct range_seg_t {
 
   // Tree is sorted by size, larger sizes at the end of the tree.
   struct shorter_t {
-    template<typename KeyType>
-    bool operator()(const range_seg_t& lhs, const KeyType& rhs) const {
+    template<typename KeyType1, typename KeyType2>
+    bool compare(const KeyType1& lhs, const KeyType2& rhs,
+                 uint64_t unit) const {
+      if (unit) {
```

Comment:
For shorter_t the ordering is:
1) aligned size
2) normal size
3) offset

This has following consequences:
A) when allocating for BlueFS 128K
aligned 128K, normal size 136K (4K pre, 4K post alignment)
has priority over
aligned 128K, normal size 140K (12K pre)
B) when allocating for BlueStore 132K
aligned 64K, normal size 132K (32K pre, 36K post alignment)
will not be found by `lower_bound()` that is in invoked with `range_t{0, 132K, 64K}`

I think that A is a just a nuisance, it does not make it worse then baseline.
The B accelerates fragmentation more then in baseline.